packages/ui-date-time-input/src/DateTimeInput/index.tsx

```diff
@@ -0,0 +1,559 @@
+        return {
+          ...this.recalculateState(this.props.value || this.props.defaultValue)
+        }
+      })
+    }
+  }
+
+  recalculateState(
+    dateStr?: string,
+    doNotChangeDate = false,
+    doNotChangeTime = false
+  ): DateTimeInputState {
+    let errorMsg: FormMessage | undefined
+    if (dateStr) {
+      const parsed = DateTime.parse(dateStr, this.locale(), this.timezone())
+      if (parsed.isValid()) {
+        if (doNotChangeTime && this.state.timeSelectValue) {
+          // There is a selected time, adjust the parsed date to its value
+          const timeParsed = DateTime.parse(
+            this.state.timeSelectValue,
+            this.locale(),
+            this.timezone()
+          )
+          parsed.hour(timeParsed.hour()).minute(timeParsed.minute())
+        }
+        if (doNotChangeDate && this.state.iso) {
+          parsed
+            .date(this.state.iso.date())
+            .month(this.state.iso.month())
+            .year(this.state.iso.year())
+        }
+        const newTimeSelectValue = this.state?.timeSelectValue
+          ? this.state.timeSelectValue
+          : this._timeInput
+              ?.getBaseDate()
+              .minute(parsed.minute())
+              .hour(parsed.hour())
+              .toISOString()
+        return {
+          iso: parsed.clone(),
+          calendarSelectedDate: parsed.clone(),
+          dateInputTextChanged: false,
+          dateInputText: parsed.format(this.dateFormat),
+          message: {
+            type: 'success',
+            text: this.props.messageFormat
+              ? parsed.format(this.props.messageFormat)
+              : parsed.format(DateTimeInput.DEFAULT_MESSAGE_FORMAT)
+          },
+          timeSelectValue: newTimeSelectValue,
+          renderedDate: parsed.clone()
+        }
+      }
+      if (dateStr.length > 0 || this.props.isRequired) {
+        const text =
+          typeof this.props.invalidDateTimeMessage === 'function'
+            ? this.props.invalidDateTimeMessage(dateStr)
+            : this.props.invalidDateTimeMessage
+        errorMsg = text ? { text, type: 'error' } : undefined
+      }
+    }
+    return {
+      iso: undefined,
+      calendarSelectedDate: undefined,
+      dateInputText: dateStr ? dateStr : '',
+      message: errorMsg,
+      renderedDate: DateTime.now(this.locale(), this.timezone()),
+      dateInputTextChanged: false
+    }
+  }
+
+  locale(): string {
+    if (this.props.locale) {
+      return this.props.locale
+    } else if (this.context && this.context.locale) {
+      return this.context.locale
+    }
+    return Locale.browserLocale()
+  }
+
+  timezone(): string {
+    if (this.props.timezone) {
+      return this.props.timezone
+    } else if (this.context && this.context.timezone) {
+      return this.context.timezone
+    }
+    return DateTime.browserTimeZone()
+  }
+
+  get dateFormat() {
+    return this.props.dateFormat ? this.props.dateFormat : 'LL'
+  }
```

Comment:
Well, the `defaultProps` will catch it if it is `undefined`. In case of `null`... hmm, not sure I'd write a guard for that, it is the users fault if they set it to null... Not to mention that it will be underlined because of the TS type.

Reply:
This is a safeguard for the case if someone sets it to `undefined/null` after creation (e.g. by accident)

Reply:
removed the check here